openfisca-web-api/input-output-data.md

```diff
@@ -1,104 +1,294 @@
 
-## Test cases
-
-A test case describes persons, entities and their associations.
+```json
+{
+  "persons": { 
+    "Ricarda": {
+      "salary": {
+        "2016-01": 3500,
+        "2016-02": 3500,
+        "2016-03": 3500,
+        "2016-04": 3500,
+        "2016-05": 3500,
+        "2016-06": 3500,
+        "2016-07": 3500,
+        "2016-08": 3500,
+        "2016-09": 4000,
+        "2016-10": 4000,
+        "2016-11": 4000,
+        "2016-12": 4000
+      }
+    },
+    "Bob": {},
+    "Bill": {},
+    "Janet": {}
+  },
+  "households": {
+    "household_1": {
+      "parents": [
+        "Ricarda", "Bob"
+      ]
+    },
+    "household_2": {
+      "parents": [
+        "Bill"
+      ],
+      "children":[
+        "Janet"
+      ],
+      "housing_occupancy_status": {
+        "2016-01": "Tenant",
+        "2016-02": "Tenant",
+        "2016-03": "Owner",
+        "2016-04": "Owner",
+        "2016-05": "Owner",
+        "2016-06": "Owner",
+        "2016-07": "Owner",
+        "2016-08": "Owner",
+        "2016-09": "Owner",
+        "2016-10": "Owner",
+        "2016-11": "Owner",
+        "2016-12": "Owner"
+      },
+      "accommodation_size": {
+        "2016-01": 57,
+        "2016-02": 57,
+        "2016-03": 57,
+        "2016-04": 57,
+        "2016-05": 57,
+        "2016-06": 57,
+        "2016-07": 57,
+        "2016-08": 57,
+        "2016-09": 57,
+        "2016-10": 57,
+        "2016-11": 57,
+        "2016-12": 57
+      }
+    }
+  }
+}
+```
+> note that because of the default value system in OpenFisca, the variables that have not been defined explicitly are either calculated or take on their default value.  
 
-Let's assume that your tax and benefit system defines a [person entity](../person,_entities,_role.md#person) named `persons`, and a [group entity]](../person,_entities,_role.md#group-entities) named `households`.
 
-Let's also assume that within a `household`, you have `parents` and `children`.
+## Computing a variable
 
-A JSON test case is an object structured this way:
-* `persons` (list of objects): defines persons with their input variables, structured this way:
-  * `id` (string): the ID of the person
-  * `<variable name (string)>` (the [value by period](#values-by-period)): an input variable
-* `households` (list of objects): the definition of the entity, structured this way:
-  * `id` (string): the ID of the entity
-  * `parents` (list of strings): a list of persons IDs referencing the ones defined under the persons key
-  * `children` (list of strings): a list of persons IDs referencing the ones defined under the persons key
-  * `<variable name (string)>` (the [value by period](#values-by-period)): an input variable
+Once you have described the situation, you can compute all variables that have a formula.
```

Comment:
You can even compute variables with no formula. It is pointless, as the result will be the default value or the input, but I'm not sure the "that have a formula" is necessary.

> Once you have described the situation, you can compute any variable.